Diagnostics/HealthChecker/DataCollection/ServerInformation/Get-Smb1ServerSettings.ps1

```diff
@@ -0,0 +1,60 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. $PSScriptRoot\..\..\..\..\Shared\Invoke-ScriptBlockHandler.ps1
+<#
+Unknown 0
+Failed to get Install Setting 1
+Install is set to true 2
+Install is set to false 4
+Failed to get Block Setting 8
+SMB1 is not being blocked 16
+SMB1 is being blocked 32
+#>
+Function Get-Smb1ServerSettings {
+    [CmdletBinding()]
+    param(
+        [string]$ServerName = $env:COMPUTERNAME,
+        [scriptblock]$CatchActionFunction
+    )
+    begin {
+        Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+        $smb1Status = 0
```

Comment:
We should either define our bit flags to make this readable (`$Failed = 1`, `$Installed = 2`, etc, then `$smb1Status += $Installed`), or not use a bitmask. Can we turn this into a PSCustomObject with a bunch of named booleans?

Reply:
@bill-long would it be better to create a `enum` class that we can instead of `$Failed = 1`, `$Installed = 2`. It would be something like this: 

```
enum Smb1Status
{
    Failed = 1
    Installed = 2
}
...

$smb1Status += [Smb1Status]::Failed 
```

Reply:
Actually the fact that you're using += is itself a problem. I even retyped it myself without thinking. This is a bitmask. We must use bitwise operators to avoid bugs. For example:

![image](https://user-images.githubusercontent.com/4518572/127695441-1fb496b4-25a6-4baa-8c5e-6c2d74ba5d5e.png)

Oops, I accidentally set my favorite day to Saturday by adding Friday twice.

Personally, I would never use a bitmask. This article sums up my opinion nicely: https://dev.to/somedood/bitmasks-a-very-esoteric-and-impractical-way-of-managing-booleans-1hlf

You're really just creating a collection of booleans, and shoving them all into one value, making them very difficult to work with, when you could just do this:

```
[PSCustomObject]@{
  Failed = $false
  Installed = $true
  ...
}
```

Reply:
Should be fully addressed in 9f1adc803ebfdaee432ad4363aa00d12591dc131

Reply:
That looks SO much better.